test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go

```diff
@@ -7,40 +7,145 @@ package integrationslogs
-			agentparams.WithFile("/etc/datadog-agent/conf.d/writeTenLogs.yaml", writeTenLogsConfig, true))))}
+			agentparams.WithAgentConfig("logs_config.integrations_logs_files_max_size: 1"))))}
+
+	suiteParams = append(suiteParams, e2e.WithDevMode())
 
 	e2e.Run(t, &IntegrationsLogsSuite{}, suiteParams...)
 }
 
 // TestWriteTenLogsCheck ensures a check that logs are written to the file ten
 // logs at a time
 func (v *IntegrationsLogsSuite) TestWriteTenLogsCheck() {
-	utils.CheckLogsExpected(v.T(), v.Env().FakeIntake, "ten_logs_service", "Custom log message", []string{"env:dev", "bar:foo"})
+	tags := []string{"foo:bar", "env:dev"}
+	yamlData, err := generateYaml("Custom log message", false, 1, 10, tags, "logs_from_integrations_source", "logs_from_integrations_service")
+	assert.NoError(v.T(), err)
+
+	v.UpdateEnv(awshost.Provisioner(awshost.WithAgentOptions(
+		agentparams.WithLogs(),
+		agentparams.WithFile("/etc/datadog-agent/conf.d/writeTenLogs.yaml", string(yamlData), true),
+		agentparams.WithFile("/etc/datadog-agent/checks.d/writeTenLogs.py", customIntegration, true))))
+
+	utils.CheckLogsExpected(v.T(), v.Env().FakeIntake, "logs_from_integrations_service", "", tags)
+}
+
+// TestIntegrationLogFileRotation ensures logs are captured after a integration
+// log file is rotated
+func (v *IntegrationsLogsSuite) TestIntegrationLogFileRotation() {
+	// Since it's not yet possible to write to the integration log file by calling
+	// the agent check command, we can test if the file rotation works using the following method:
+	// 1. Send logs until to the logs from integrations launcher until their
+	// cumulative size is greater than integration_logs_files_max_size (for logs
+	// of size 256kb, this will be four logs given the max size of 1mb)
+	// 2. Put a counter in the logs that increases by 1 every time the check is
+	// called, then check that each log's counter is 1 more than the previous log
+
+	tags := []string{"test:rotate"}
+	yamlData, err := generateYaml("a", true, 1024*230, 1, tags, "rotation_source", "rotation_service")
+	assert.NoError(v.T(), err)
+
+	v.UpdateEnv(awshost.Provisioner(awshost.WithAgentOptions(
+		agentparams.WithLogs(),
+		agentparams.WithFile("/etc/datadog-agent/conf.d/rotation.yaml", string(yamlData), true),
+		agentparams.WithFile("/etc/datadog-agent/checks.d/rotation.py", customIntegration, true))))
+
+	// Check each log is received and is unique
+	prevLogCount := -1
+	for i := 0; i < 5; i++ {
+		assert.EventuallyWithT(v.T(), func(c *assert.CollectT) {
+			logs, err := utils.FetchAndFilterLogs(v.Env().FakeIntake, "rotation_service", ".*counter: \\d+.*")
+			assert.NoError(c, err)
+
+			if assert.NotEmpty(c, logs) && len(logs) >= i+1 {
+				log := logs[i]
+				regex := regexp.MustCompile(`counter: (\d+)`)
+				matches := regex.FindStringSubmatch(log.Message)
+				assert.Greater(v.T(), len(matches), 1, "Did not find count in log")
```

Comment:
```suggestion
				assert.Greater(c, len(matches), 1, "Did not find count in log")
```
Or even better use [assert.NotEmpty](https://pkg.go.dev/github.com/stretchr/testify/assert#NotEmpty).

Reply:
changed!